adding Ingridients controller and js logic, and java doc for admincon…

Mads505m opened the pull request:
Change some html (hidden inputfields) Used it in js and controller to connect inputs to the js and then add the inputted the data to the database, in which the function was already made.

Mads505m committed:
adding Ingridients controller and js logic, and java doc for admincontroller

Change some html (hidden inputfields) Used it in js and controller to connect inputs to the js and then add the inputted the data to the database, in which the function was already made.

Emilkroldrup commented on src/main/resources/static/javascript/adminPage.js:
Kan godt være jeg lidt lost her. Men giver det mening og lave alt dette i js og ikke java? da vi bliver bedømt på java code og ikke js? Js fungere godt I know, men det mere om det måske var vigtigere og mappe direkte til thymeleaf og java, i stedet for igennem et script ^^ ellers ser det sku fornuftigt ud

Mads505m replied:
Det rigtigt, men det var det peter allerede havde gjort

Emilkroldrup reviewed (changes requested):
>

Emilkroldrup committed:
Merge branch 'develop' into adminpage

Mads505m committed:
Ingredient and recipe added to ingredientrecipe, quantity being able to be added, exceptions.

Added duplicate exception to saverecipe.
Made recipe name unique
Made get recipebyid function.
Changed the genericrepository gettable by name to be more inclusive of classes that dont have the same name as database.
Changed up the constructor in recipeingredients.
Made recipeingredientimpl and interface.
Made recipeingredients save function.
Added recepi id to recepi.
changed users name of function parameter to user.
fixed findbyproperty functions in reciperepository and ingredient repository.

Collabing with Peter, for a good amount of stuff.

Co-Authored-By: Spi0nDK <73273853+Spi0nDK@users.noreply.github.com>

Emilkroldrup commented on src/main/java/minkostplan/application/DBcontroller/user/UserRepositoryImpl.java:
og ændre til user i stedet for users, laver ikke fejl nogle steder aka som i security?

Emilkroldrup commented on src/main/java/minkostplan/application/UIcontroller/AdminController.java:
```diff
@@ -30,13 +37,13 @@ public class AdminController {
     @GetMapping("/admin")
     public String adminPage(Model model) {
         Authentication auth = SecurityContextHolder.getContext().getAuthentication();
-        String email = auth.getName(); // get logged-in username
```
Helle dette statement er lavet om i en anden PR

Mads505m committed:
Merge branch 'develop' into adminpage
